### DISCUSSION TITLE
SPXSpeechSynthesizer output volume and speaker configuration on iOS

### DISCUSSION THREAD

**Discussion by lukaspetrik** [MacPaw/OpenAI#129] [Q&A]
Hello, is there a way to change output speakers and sound volume of SPXSpeechSynthesizer? It's only outputting audio in the top speaker on iPhone for me - similar to when you have a phone call. I would like the audio to be played on bottom speakers - loud speakers.

**Comment by kalafus** [comment]
There's no SPXSpeechSythesizer in Xcode documentation... apparently that's part of Microsoft Speech SDK.

Maybe it's using `.voiceChat` mode? According to Xcode documentation, `AVAudioSession.Mode.voiceChat` is "A mode that indicates that your app is performing two-way voice communication, such as using Voice over Internet Protocol (VoIP)." Seems like that would disable the speaker on the microphone side. Maybe there's a passthrough setting somewhere in the Microsoft Speech SDK, or maybe it's a bug in the SDK 🤷 

```
import AVFoundation
class SpeakerController {

    var isUsingEarpiece = false

    func toggleAudioOutput() {
        let audioSession = AVAudioSession.sharedInstance()

        do {
            try audioSession.setCategory(.playAndRecord, mode: .voiceChat, options: .defaultToSpeaker)

            if isUsingEarpiece {
                try audioSession.overrideOutputAudioPort(.none)
            } else {
                try audioSession.overrideOutputAudioPort(.speaker)
            }

            try audioSession.setActive(true)

            isUsingEarpiece = !isUsingEarpiece
        } catch {
            print("Error toggling audio output: \(error.localizedDescription)")
        }
    }
}
```
